File: cn/java-spring-webflux-demo/src/main/java/com/itangcent/springboot/webflux/demo/controller/SecurityCtrl.java
package com.itangcent.springboot.webflux.demo.controller;


import com.itangcent.common.model.TestJsonFieldBean;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;
import reactor.core.publisher.Mono;

import javax.annotation.security.RolesAllowed;


/**
 * 安全控制器
 * 用于演示基于角色的访问控制功能
 */
@RestController
@RequestMapping(value = "/test/security")
public class SecurityCtrl {

    /**
     * 测试角色权限控制
     * 演示使用@RolesAllowed注解进行访问控制
     * 只有具有admin角色的用户才能访问此接口
     *
     * @param testJsonFieldBean 测试数据对象
     * @return 处理结果
     */
    @RolesAllowed("admin")
    @RequestMapping("/rolesAllowed")
    public Mono<TestJsonFieldBean> rolesAllowed(
            @RequestBody TestJsonFieldBean testJsonFieldBean) {
        return Mono.just(testJsonFieldBean);
    }

}

File: en/kotlin-spring-demo/src/main/kotlin/com/itangcent/springboot/kotlin/demo/controller/SecurityCtrl.kt
package com.itangcent.springboot.kotlin.demo.controller

import com.itangcent.common.annotation.RolesAllowedInJava
import com.itangcent.common.model.TestJsonFieldBean
import org.springframework.web.bind.annotation.RequestBody
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RestController
import javax.annotation.security.RolesAllowed

/**
 * Security Controller
 * Used to demonstrate Spring Security's role-based access control functionality
 */
@RestController
@RequestMapping("/test/security")
class SecurityCtrl {

    /**
     * Role-based Access Control Using Standard Annotation
     * Demonstrates using @RolesAllowed annotation for permission control
     *
     * @param testJsonFieldBean Test data object
     * @return Returns the original test object
     * @roles admin Administrator role required for access
     */
    @RolesAllowed("admin")
    @RequestMapping("/rolesAllowed")
    fun rolesAllowed(
            @RequestBody testJsonFieldBean: TestJsonFieldBean): TestJsonFieldBean {
        return testJsonFieldBean
    }

    /**
     * Role-based Access Control Using Custom Java Annotation
     * Demonstrates using custom @RolesAllowedInJava annotation for permission control
     *
     * @param testJsonFieldBean Test data object
     * @return Returns the original test object
     * @roles admin Administrator role required for access
     */
    @RolesAllowedInJava(value = "admin", open = false)
    @RequestMapping("/rolesAllowed/java")
    fun rolesAllowedJava(
            @RequestBody testJsonFieldBean: TestJsonFieldBean): TestJsonFieldBean {
        return testJsonFieldBean
    }
}
